hooks/useFilteredQuestChains.ts

```diff
@@ -19,21 +21,24 @@ export const useQuestChainSearchForAllChains = (
     (async () => {
       try {
         setFetching(true);
-        const allResults = await Promise.all(
-          SUPPORTED_NETWORKS.map(async chainId =>
-            graphql.getQuestChainsFromSearch(chainId, search),
-          ),
-        );
+        let chains: graphql.QuestChainInfoFragment[] = [];
+
+        if (network && isSupportedNetwork(network)) {
+          chains = await graphql.getQuestChainsFromFilters(network, filters);
+        } else {
+          const allResults = await Promise.all(
+            SUPPORTED_NETWORKS.map(async n =>
+              graphql.getQuestChainsFromFilters(n, filters),
+            ),
+          );
+          chains = allResults.reduce((t, a) => {
+            t.push(...a);
+            return t;
+          }, []);
```

Comment:
I just realised based on the sort options we will have to sort this again when there are more than one network that we are searching for multiple networks.